core/resources/directional_light.glsl

```diff
@@ -0,0 +1,29 @@
+struct DirectionalLight {
+	vec4 ambient;
+	vec4 diffuse;
+	vec4 specular;
+
+	vec3 direction;
+};
+
+void calculateLight(in DirectionalLight _light, in vec3 _eye, in vec3 _eyeToPoint, in vec3 _normal, inout vec4 _ambient, inout vec4 _diffuse, inout vec4 _specular){
+	#ifdef MATERIAL_AMBIENT
```

Comment:
Minor point but we might want to namespace these to avoid possible collisions with user-defined #defines. Like `TANGRAM_MATERIAL_AMBIENT` or something.